### ISSUE TITLE
Inference tends to not stop easily

### ISSUE THREAD

**Issue by wdp-007** [CSfufu/Revisual-R1#5]
I am using the code you provided to replicate the Code Start training. During my experiment, I randomly sampled 40k examples from the DeepMath dataset and trained for 5 epochs. However, I encountered an issue during testing where the model's inference tends to not stop easily, leading to the truncation of many final outputs. I set the maximum output length to 16k, which resulted in lower mathv scores than expected. Have you encountered this issue, and do you have any suggestions for addressing it?

Training Configuration:
```
dataset: deepmath
output_dir: 7b-instruct
run_name: REVISUAL_COLD_START 
### model
model_name_or_path: qwen2_5vl-7b-release-instruct
trust_remote_code: true
### method
stage: sft
do_train: true
finetuning_type: full
# deepspeed: examples/deepspeed/ds_z3_offload_config.json
deepspeed: cold_start/examples/deepspeed/ds_z3_offload_config.json

### dataset
# dataset: bespoke_cot
dataset_dir: ./deepmath_dataset
template: qwen2_vl
# cutoff_len: 32768
cutoff_len: 24000
overwrite_cache: true
preprocessing_num_workers: 1024

### output
logging_steps: 1
save_steps: 150
save_total_limit: 10
save_only_model: false
save_strategy: 'steps'
plot_loss: true
overwrite_output_dir: false
report_to: wandb


### train
per_device_train_batch_size: 1
gradient_accumulation_steps: 16
learning_rate: 1.0e-5
# weight_decay: 1.0e-4
num_train_epochs: 5
lr_scheduler_type: cosine
warmup_ratio: 0.05
bf16: true
ddp_timeout: 540000000
# resume_from_checkpoint: true 

### eval
val_size: 0
per_device_eval_batch_size: 1
eval_strategy: 'no'
eval_steps: 10000000

###tokenizer
# add_tokens: <think>,</think>,<answer>,</answer>
# skip_special_tokens: false
# resize_vocab: true


```

**Comment by CSfufu** [comment]
Thanks for your detailed reproduction setup!

Yes, we’ve seen similar behavior when the model hasn’t been trained sufficiently, especially in the early stages of cold-start fine-tuning. Based on your config, it looks like your `cutoff_len` is set to 24000, and you’re using 40k examples trained over 5 epochs — that’s definitely on the low side.

In our experiments, using a `cutoff_len` of 32768 and training for at least 3k+ steps was essential. More importantly, the loss needs to reduce to around 0.35–0.4 to ensure the model has learned to generate well-structured and bounded outputs. Otherwise, it tends to over-generate or fail to stop, which leads to truncation issues and degraded MathV scores, as you observed.

Let me know how it goes — happy to help troubleshoot further!

**Comment by wdp-007** [comment]
Thanks for your insight! I followed the paper's configuration and trained with 40k data for 5 epochs on 8 A100 GPUs, resulting in about 1565 steps. The loss is already within the 0.36–0.4 range. So, are you suggesting that during the cold-start phase, it's essential to train on 40k deepmath data for more than 5 epochs, possibly overfitting even up to 10+ epochs to ensure sufficient training?

**Comment by CSfufu** [comment]
Thanks for your interest and for sharing your results!

A MathVision score of around 24 after 1000 steps likely suggests that training is not yet sufficient. In our experiments, improvements continue steadily up to and beyond 3000 steps, so further training should help boost performance. Another important factor is evaluation: please ensure that prefetch is properly enabled during testing, as missing it can significantly impact results. Lastly, the quality and composition of your dataset may also play a role—make sure it's consistent with the setup we used (e.g., filtering for well-formed and diverse samples).
